fetch/fetch.go

```diff
@@ -268,16 +277,22 @@ func (f *Fetch) loop() {
 	for {
 		select {
 		case req := <-f.requestReceiver:
-			// group requests by hash
+			sendNow := req.priority > Low
 			f.activeReqM.Lock()
-			f.activeRequests[req.hash] = append(f.activeRequests[req.hash], &req)
+			// group requests by hash
+			if sendNow {
```

Comment:
what happens if we have one hash sent with high priority and then with low priority, will this will still result in 2 calls made to other peers?

Reply:
prior to the change, for a high priority hash req1, we will send out two requests, one now, and one within the batch.
now it just send out a request for req1 now, and not in the batch. 

but if someone send in req1 (high priority) and req2 (low priority), req1 will be sent right away and req2 will be batched up.

in short. old behavior: high priority req got sent twice. new behavior, high priority got sent just once.

Reply:
>but if someone send in req1 (high priority) and req2 (low priority), req1 will be sent right away and req2 will be batched up.

we can even have this case optimised if we check if there are pending requests with this hash right?

Reply:
good idea. i think that requires a bigger code change and more tests to change both high and low priority requests. i will do it in the next PR together with reduce logging